mentoring-app/src/app/courses-page/course/edit-course/edit-course.component.ts

```diff
@@ -10,26 +12,33 @@ import { ActivatedRoute, Router } from '@angular/router';
 })
 export class EditCourseComponent implements OnInit {
 
-  courseInfo: CourseClass;
+  currentCourse: CourseClass;
+  currentCourseInfo: CourseClass;
 
   constructor(
-    private coursesService: CoursesService,
     private route: ActivatedRoute,
-    private router: Router) { }
+    private router: Router,
+    private store: Store<CoursesState>) { }
 
   ngOnInit() {
     this.route.paramMap.subscribe((params) => {
-      this.coursesService.getCourse(parseInt(params.get('id'), 10))
-        .subscribe(course => this.courseInfo = course);
+      this.store.dispatch(new GetCourse(parseInt(params.get('id'), 10)));
```

Comment:
no need to reach out a server each time, when you go from courses-list to edit course page, the course should already be in your store. You should handle both cases: 1) when user goes from courses-list then course should be taken from store; 2) when user reloads edit page and there is no data in the store then you should fetch the course from server. Simple solution would be combine the two approaches - if no data in the store only then fetch it from the server. You could create a resolver for that for example just to not keep the logic in the component.